ext/filter/filter.c

```diff
@@ -660,8 +660,8 @@ PHP_FUNCTION(filter_input)
 	ZEND_PARSE_PARAMETERS_END();
 
 	if (!PHP_FILTER_ID_EXISTS(filter)) {
-		php_error_docref(NULL, E_WARNING, "Unknown filter with ID " ZEND_LONG_FMT, filter);
-		RETURN_FALSE;
+		zend_argument_value_error(3, "must be a valid filter ID");
+		RETURN_THROWS();
```

Comment:
These warnings were only added in PHP 8 (previously silently returned false), so I don't think we should make them throw now :)

Reply:
Oh noo! :) I didn't understand why I hadn't promoted them... But now, it came to my mind that you added them. I'll revert these then :(